features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Apis/Details/AsyncApiConsole/WebhookSubscriptionUI.jsx

```diff
@@ -0,0 +1,270 @@
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import Typography from '@material-ui/core/Typography';
+import AccordionActions from '@material-ui/core/ExpansionPanelActions';
+import Button from '@material-ui/core/Button';
+import CopyToClipboard from 'react-copy-to-clipboard';
+import TextField from '@material-ui/core/TextField';
+import Alert from 'AppComponents/Shared/Alert';
+import { makeStyles } from '@material-ui/core/styles/index';
+import Grid from '@material-ui/core/Grid';
+import Radio from '@material-ui/core/Radio';
+import { RadioGroup } from '@material-ui/core';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import { injectIntl } from 'react-intl';
+
+const useStyles = makeStyles((theme) => (
+    {
+        bootstrapRoot: {
+            padding: 0,
+            'label + &': {
+                marginTop: theme.spacing(1),
+            },
+        },
+        bootstrapInput: {
+            borderRadius: 4,
+            backgroundColor: theme.palette.common.white,
+            border: '1px solid #ced4da',
+            padding: '5px 12px',
+            marginTop: '11px',
+            marginBottom: '11px',
+            width: '100%',
+            transition: theme.transitions.create(['border-color', 'box-shadow']),
+            '&:focus': {
+                borderColor: '#80bdff',
+                boxShadow: '0 0 0 0.2rem rgba(0,123,255,.25)',
+            },
+            fontSize: 12,
+        },
+        bootstrapCurl: {
+            borderRadius: 4,
+            backgroundColor: '#41444e',
+            color: '#fff',
+            border: '1px solid #ced4da',
+            padding: '5px 12px',
+            marginTop: '11px',
+            marginBottom: '11px',
+            width: '100%',
+            transition: theme.transitions.create(['border-color', 'box-shadow']),
+            '&:focus': {
+                borderColor: '#80bdff',
+                boxShadow: '0 0 0 0.2rem rgba(0,123,255,.25)',
+            },
+            fontSize: 12,
+            fontFamily: 'monospace',
+            fontWeight: 600,
+        },
+        subscriptionSummary: {
+            backgroundColor: '#89b7d1',
+            maxHeight: '40px',
+            borderColor: '#80bdff',
+            '&$expanded': {
+                maxHeight: '40px',
+            },
+        },
+        subscription: {
+            paddingBottom: '10px',
+        },
+    }
+));
+
+function reducer(state, { field, value }) {
+    return { ...state, [field]: value };
+}
+
+function WebhookSubscriptionUI(props) {
+    const classes = useStyles();
+    const { generateGenericWHSubscriptionCurl, topic, intl } = props;
+    const initialSubscriptionState = {
+        topic: topic.name,
+        secret: null,
+        lease: 50000,
+        mode: 'subscribe',
+        callback: null,
+    };
+    const [curl, setCurl] = useState(generateGenericWHSubscriptionCurl(initialSubscriptionState));
```

Comment:
This kind of complex state favor to use `useReducer`

> useReducer is usually preferable to useState when you have complex state logic that involves multiple sub-values . . .

~[React Docs](https://reactjs.org/docs/hooks-reference.html#usereducer)

Reply:
ahh ok this state only holds the curl generated from the `initialSubscriptionState` 🤓